hoodie-common/src/main/avro/HoodieArchivedMetaEntry.avsc

```diff
@@ -0,0 +1,59 @@
+{
+   "type":"record",
+   "name":"HoodieArchivedMetaEntry",
+   "namespace":"com.uber.hoodie.avro.model",
+   "fields":[
+      {
+         "name":"hoodieCommitMetadata",
+         "type":[
+            "null",
+            "HoodieCommitMetadata"
+         ],
+         "default": "null"
+      },
+      {
+         "name":"hoodieCleanMetadata",
+         "type":[
+            "null",
+            "HoodieCleanMetadata"
+         ],
+         "default": "null"
+      },
+      {
+         "name":"hoodieCompactionMetadata",
+         "type":[
+            "null",
+            "HoodieCompactionMetadata"
+         ],
+         "default": "null"
+      },
+      {
+         "name":"hoodieRollbackMetadata",
+         "type":[
+            "null",
+            "HoodieRollbackMetadata"
+         ],
+         "default": "null"
+      },
+      {
+         "name":"hoodieSavePointMetadata",
+         "type":[
+            "null",
+            "HoodieSavepointMetadata"
+         ],
+         "default": "null"
+      },
+      {
+         "name":"commitTime",
+         "type":["null","string"]
+      },
+      {
+         "name":"actionType",
```

Comment:
Unable to make this optional because of the following bug in avro code generation : 

The Enum class from the code generation results in the following class : 
@SuppressWarnings("all")
@org.apache.avro.specific.AvroGenerated
public enum ActionType { 
  commit, savepoint, compaction, clean, rollback  ;
  public static final org.apache.avro.Schema SCHEMA$ = new org.apache.avro.Schema.Parser().parse("{\"type\":\"enum\",\"name\":\"ActionType1\",\"namespace\":\"com.uber.hoodie.avro.model\",\"symbols\":[\"commit\",\"savepoint\",\"compaction\",\"clean\",\"rollback\"]}");
  public static org.apache.avro.Schema getClassSchema() { return SCHEMA$; }
}

Notice that the class does not extend GenericRecord.EnumSymbol which it should to say that this class is of type EnumSymbol. 
Now when we make this optional, the GenericDatumWriter fails because it hits a resolveUnion() piece of code which expects ActionType to have extended GenericData.EnumSymbol but the code generated class hasn't and throws an exception. I can explain in more detail if this is unclear. 
For now, I'm leaving this as non-optional. If we want, we can make this as a string instead of Enum, that way it's optional.
Bottom line, there is a bug in the Avro Code generation. I'll try to find a JIRA for the same.